src/main/java/org/jabref/logic/openoffice/backend/Backend52.java

```diff
@@ -0,0 +1,465 @@
+        PropertyVetoException,
+        IllegalTypeException {
+
+        Objects.requireNonNull(pageInfos);
+        if (pageInfos.size() != citationKeys.size()) {
+            throw new IllegalArgumentException("pageInfos.size != citationKeys.size");
+        }
+
+        // Get a new refMarkName
+        Set<String> usedNames = new HashSet<>(this.citationStorageManager.nrmGetUsedNames(doc));
+        String xkey = (citationKeys.stream().collect(Collectors.joining(",")));
+        String refMarkName = Codec52.getUniqueMarkName(usedNames, xkey, citationType);
+
+        CitationGroupId cgid = new CitationGroupId(refMarkName);
+
+        final int nCitations = citationKeys.size();
+        final int last = nCitations - 1;
+
+        // Build citations, add pageInfo to each citation
+        List<Citation> citations = new ArrayList<>(nCitations);
+        for (int i = 0; i < nCitations; i++) {
+            Citation cit = new Citation(citationKeys.get(i));
+            citations.add(cit);
+
+            Optional<OOText> pageInfo = PageInfo.normalizePageInfo(pageInfos.get(i));
+            switch (dataModel) {
+            case JabRef52:
+                if (i == last) {
+                    cit.setPageInfo(pageInfo);
+                } else {
+                    if (pageInfo.isPresent()) {
+                        LOGGER.warn("dataModel JabRef52"
+                                    + " only supports pageInfo for the last citation of a group");
+                    }
+                }
+                break;
+            case JabRef60:
+                cit.setPageInfo(pageInfo);
+                break;
+            }
+        }
+
+        /*
+         * Apply to document
+         */
+        boolean withoutBrackets = (citationType == CitationType.INVISIBLE_CIT);
+        NamedRange namedRange =
+            this.citationStorageManager.nrmCreate(doc, refMarkName, position, insertSpaceAfter,
+                                                  withoutBrackets);
+
+        switch (dataModel) {
+        case JabRef52:
+            Optional<OOText> pageInfo = PageInfo.normalizePageInfo(pageInfos.get(last));
+
+            if (pageInfo.isPresent()) {
+                String pageInfoString = OOText.toString(pageInfo.get());
+                UnoUserDefinedProperty.setStringProperty(doc, refMarkName, pageInfoString);
+            } else {
+                // do not inherit from trash
+                UnoUserDefinedProperty.removeIfExists(doc, refMarkName);
+            }
+            CitationGroup cg = new CitationGroup(OODataModel.JabRef52,
+                                                 cgid,
+                                                 citationType, citations,
+                                                 Optional.of(refMarkName));
+            this.cgidToNamedRange.put(cgid, namedRange);
+            return cg;
+        default:
+            throw new IllegalStateException("Backend52 requires JabRef52 dataModel");
+        }
```

Comment:
> Is this something to be completed in one of the followup prs?

Not in the oobranch-[A-J] series.

Changed to 
```java
        case JabRef60:
            throw new IllegalStateException("createCitationGroup for JabRef60 is not implemented yet");
        default:
            throw new IllegalStateException("Unhandled dataModel in Backend52.createCitationGroup");
```
to align with other parts.


My idea here was to eventually separate Backend52 and Backend60 (or whatever it is going to be called) behind a hopefully common interface. In this line of thought I considered the `case JabRef60:` branches temporary
here, only existing to make it easier to compare with the Jabref52/Backend52 parts.

Backend60 (or handling Jabref60) is not finished (and not moved to a separate class).
Suggestions seemed to point towards minimally changing the encoding, and providing only an upgrade possibility from the old encoding (not editing or exporting to older representation).

This would allow us to drop Jabref52-specific handling from at least "Separate citations" and "Merge citations",
and move back code from Backend to the action implementation.

On the other hand there are several problems with the encoding used and I am afraid releasing a new backend
for each fix might result in too many variants to handle.

@koppor 
Lately [koppor suggested](https://github.com/JabRef/jabref/issues/2146#issuecomment-891385140)  to 
"use the same format as Zotero? So that OO documents can be used with Zotero and JabRef interchangeably"
and "My proposal would be: Switch to CSL only for JabRef 6.0"

This brings further changes affecting the representation in view. 
In short, I am not sure what is the next step after oobranch-J. Implement Jabref60/Backend60 with minimal changes,
or extend internal handling first to cover all parts of Zotero (prefix, suffix, ...) internally, then try to use zotero-like encoding in the document?

Reply:
ah ok, I think this should be discussed explicitly. But discussing the future development of this should not stop us from integrating this oo pr series.